### PR TITLE
MTDSA-15111: CL102 Feature Switch in V1 Create/Amend Pension Charges

### PR DESCRIPTION
**Pull request by tom-lloyd-hmrc**
This PR uses the `cl102` feature switch to decide whether to move `isAnnualAllowanceReduced`, `moneyPurchasedAllowance` and `taperedAnnualAllowance` from the `pensionSavingsTaxCharges` object to the `pensionContributions` object.

When it is disabled, they are kept where they are. When it is enabled they are moved.

There is an issue where, when we move the fields, the `pensionContributions` object might not exist (it is optional) but it is not possible to create the object, as it has other mandatory fields that we do not know the value of. I have spoken to Fatih, who has said that this is an edge case and we should return a 500 Internal Server Error in this scenario.

### CODE REVIEW

File: test/v1/services/AmendPensionsChargesServiceSpec.scala [hunk 1]
```diff
@@ -54,7 +64,7 @@ class AmendPensionsChargesServiceSpec extends ServiceSpec {
     }
 
     "return that wrapped error as-is" when {
-      "the connector returns an outbound error" in new Test {
+      "the connector returns an outbound error" in new Cl102Disabled {
         val someError          = MtdError("SOME_CODE", "some message", BAD_REQUEST)
```
Comment (GazalNebHMRC) [inline test/v1/services/AmendPensionsChargesServiceSpec.scala:68]: Please add type-annotations for public vals in this file.

**Review by GazalNebHMRC — CHANGES_REQUESTED** [review]

**Review by GazalNebHMRC — APPROVED** [review]
